### ISSUE TITLE
[Awaiting Miro Payment] [$1000] Timezone select page does not navigate back to timezone initial page after user changes timezone to automatic from another device

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#24516]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login the same account on 2 devices
2. From device A, go to the timezone details page and turn off automatic timezone selection
3. From device B, go to the timezone select page.
4. From device A, turn on automatic timezone selection.
5. Notice in device B, user remains on the timezone select page, and if they come back to timezone details page, they cant go back to the select page because their timezone is automatic
## Expected Result:
User should be redirected back to timezone details page if they update their timezone to automatic
## Actual Result:
User is not redirected back to timezone details page if they update their timezone to automatic
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.53-1
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/38435837/4b0c8051-d352-41b9-8ee6-cb5973a16f2b

https://github.com/Expensify/App/assets/38435837/952d95dd-943b-46f0-87de-0203d2b42112

**Expensify/Expensify Issue URL:**
**Issue reported by:** @huzaifa-99
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1691280927472319

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b0a8cd3364d19ad6</li>
        <li>Upwork Job ID: 1691038383142993920</li>
        <li>Last Price Increase: 2023-08-14</li>
<li>Automatic offers: </li>
<ul>
<li>bernhardoj | Contributor | 26131162</li>
<li>huzaifa-99 | Reporter | 26131163</li>
</ul></li>
    </ul>
</details>

**Comment by m-natarajan** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
We want the user to navigate back to the initial timezone page from the select page, if they change their timezone to automatic from another device.
### What is the root cause of that problem?
We are not navigating the user back to initial timezone page when they change their timezone to automatic from another device.
### What changes do you think we should make in order to solve the problem?
We are already listening to the `props.currentUserPersonalDetails` in `TimezoneSelectPage` so we just need to navigate back to timezone initial page from the timezone select page whenever
1. User updates their timezone to automatic from another device
2. User loads the timezone select page URL `/settings/profile/timezone/select`
something like this pseudocode [here](https://github.com/Expensify/App/blob/9294e9eacd53ba2ba805c6e33f18c796ccc3306d/src/pages/settings/Profile/TimezoneSelectPage.js#L33) and [here](https://github.com/Expensify/App/blob/9294e9eacd53ba2ba805c6e33f18c796ccc3306d/src/pages/settings/Profile/TimezoneSelectPage.js#L48)
```
if(timezone.automatic) {
    Navigation.goBack(ROUTES.SETTINGS_TIMEZONE);
    return
}
```
### What alternative solutions did you explore? (Optional)
N/A

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Timezone select page does not navigate back to timezone initial page after user changes timezone to automatic from another device
### What is the root cause of that problem?
User can open select timezone page by link normally without any handling to hide this when the timezone is automatic.
### What changes do you think we should make in order to solve the problem?
We need to wrap page content with `FullPageNotFoundView`
```javascript
<ScreenWrapper includeSafeAreaPaddingBottom={false}>
  <FullPageNotFoundView shouldShow={this.timezone.automatic}> 
     ...
  </FullPageNotFoundView>
</ScreenWrapper>
```
We can also display specific message in FullPageNotFoundView like you can't select timezone
### What alternative solutions did you explore? (Optional)
N/A

**Comment by trjExpensify** [comment]
I don't think anyone is going to really run into the scenario in the OP in practice. That said, if I'm following it correctly and there's a regression insofar as being able to load the list of timezones from a URL and a fix here resolves that as well, let's go ahead.

**Comment by trjExpensify** [comment]
Let us know what you think about these proposals, @0xmiroslav!

**Comment by mountiny** [comment]
Thanks Miroslav

@bernhardoj can you make the pr please

**Comment by mountiny** [comment]
Yeah I agree, bonus can be applied here cc @trjExpensify

**Comment by trjExpensify** [comment]
Sounds good. :)

**Comment by mountiny** [comment]
@bernhardoj @0xmiroslav seems like the automation failed here and this is ready for payment, can you summarize the prices here please?

**Comment by trjExpensify** [comment]
bump @bernhardoj @0xmiroslav

**Comment by trjExpensify** [comment]
Thanks, super helpful! So confirming payments due as follows: 

- no regression deductions
- #urgency bonus is due 
- bounty prices pre-date the price change


$250 to @huzaifa-99 for the bug report
$1,500 to @0xmiroslav for the C+ review & #urgency bonus 
$1,500 to @bernhardoj for the fix & #urgency bonus 

Offers sent!

**Comment by trjExpensify** [comment]
@bernhardoj - paid!
@huzaifa-99 - paid! 
@0xmiroslav - waiting on you now.

**Comment by trjExpensify** [comment]
@0xmiroslav like the other issue, if you're tracking this somewhere else. I'm going to close it!
